### PR TITLE
Add linked plotting example to bokehjs

### LINKED ISSUE #4159: Add linked plotting example to bokehjs

### PR DESCRIPTION
**Pull request by mattpap**
- [x] create an example based on `linked_brushing.py` and `linked_panning.py`
- [x] make axes and grids appear for data ranges
- [x] add `GridPlot(children)` syntax (on top of `new GridPlot({children: children})`)
- [x] allow to reuse the same data source; create unique indexed field names to avoid conflicts

fixes #4159

### CODE REVIEW

**Comment by mattpap** [comment]
Heh, "linked bushing".

File: bokehjs/examples/linked/linked.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  <head>
+    <meta charset="utf-8">
+    <title>Linked Brushing &amp; Panning</title>
+    <link rel="stylesheet" href="../../build/css/bokeh.css" type="text/css" />
+    <script type="text/javascript" src="../../build/js/bokeh.js"></script>
+  </head>
+  <body>
+    <script type="text/javascript" src="linked.js"></script>
+  </body>
+</html>
```
Comment (bryevdv) [inline bokehjs/examples/linked/linked.html]: side question: is it possible and/or desirable to automate these boilerplate HTML files?

**Comment by bryevdv** [comment]
LGTM

File: bokehjs/examples/linked/linked.html (hunk 1 above)
Comment (mattpap) [inline bokehjs/examples/linked/linked.html]: Currently `anscombe` example has additional content to show how to embed without `plt.show()`. There may be new examples needing custom content as well. We will see, maybe some can be auto-generated.

Comment (bryevdv) [inline bokehjs/examples/linked/linked.html]: OK, it's not a big deal for now just wondering.

**Pull request merged**